### ISSUE TITLE
"Unreachable code detected (7027)" false positive in 3.7

### ISSUE THREAD

**Issue by iyegoroff** [microsoft/TypeScript#34879]
**TypeScript Version:**  3.7-beta, 3.7.1-rc, Nightly

**Search Terms:** declare never unreachable 7027

**Code**

```ts
declare function foo(): never

function bar() {
    foo()

    return 1
}
```

**Expected behavior:** no errors

**Actual behavior:** received error: "Unreachable code detected.(7027)"

**Playground Link:**  https://www.typescriptlang.org/play/index.html?ts=3.8.0-dev.20191101#code/PTAEDMFcDsGMBcCWB7aFnIBQEpQG8AoEUEkgJwFN5Iy0BGUAQwGdRoKA3CsosAXwIEAJhVgAbRpQgwEKNOAw4AXG07dBUOElSgARpJz4CpdFmyCTlarVB0CAgkA

**Related Issues:**  -

**Comment by MartinJohns** [comment]
I would say this works as intended. The function `foo` declares that it will **never** return (by having `never` as the return-type). So the compiler correctly assumes that any code that comes after calling this method won't ever run.

Related issue: #28859

**Comment by fatcerberus** [comment]
https://devblogs.microsoft.com/typescript/announcing-typescript-3-7-rc/#better-support-for-never-returning-functions

**Comment by iyegoroff** [comment]
@MartinJohns , @fatcerberus , thanks for the answers! 

> Why do you consider this a false positive?

Because it wasn't an error in 3.6 and https://www.typescriptlang.org/play/index.html?ts=3.8.0-dev.20191101&ssl=3&ssc=2&pln=1&pc=1#code/GYVwdgxgLglg9mABMOcAUBKRBvAUIgxAJwFMoQikBGRAQwGdEwSA3Eo3AX111ElgSIARrSKYc+QinQYehYmQrUuPIA also doesn't produce any errors
